### PR TITLE
Add signing typed data to keyrings

### PR DESCRIPTION
**Pull request by wojciech-turek**
Adds a keyring function that will allow signing typed data, the request will come from the ui through main.

### CODE REVIEW

File: background/services/keyring/index.ts [hunk 1]
```diff
@@ -331,6 +332,21 @@ export default class KeyringService extends BaseService<Events> {
     return newAddress
   }
 
+  /**
+   * Find keyring associated with an account.
+   *
+   * @param account - the account desired to search the keyring for.
+   */
+  findKeyring = async (account: HexString): Promise<HDKeyring> => {
```
Comment (mhluongo) [inline background/services/keyring/index.ts:340]: Two things here

1. Why are we avoiding normal async method syntax?
2. This should be private using `#` We don't want key material to leave the service outside a very tightly defined API.

**Review by mhluongo — CHANGES_REQUESTED** [review]

File: background/services/keyring/index.ts (hunk 1 above)
Comment (Shadowfiend) [reply background/services/keyring/index.ts:340]: To be a little more precise here, this should just be:

```suggestion
  async #findKeyring(account: HexString): Promise<HDKeyring> {
```

File: background/services/keyring/index.ts [hunk 2]
```diff
@@ -397,6 +408,27 @@ export default class KeyringService extends BaseService<Events> {
     return signedTx
   }
 
+  async signTypedData({
```
Comment (Shadowfiend) [inline background/services/keyring/index.ts:411]: Let's give this some docs, particularly calling out the use of the latest `signTypedData` version defined in EIP-712, and link to https://eips.ethereum.org/EIPS/eip-712 .

File: background/types.ts [hunk 3]
```diff
@@ -50,3 +52,17 @@ export enum KeyringTypes {
...
 }
+
+export type EIP712DomainType = {
+  name?: string
+  version?: string
+  chainId?: number
+  verifyingContract?: HexString
+}
+
+export type TypedData = {
```
Comment (Shadowfiend) [inline background/types.ts:63]: Let's call this EIP712TypedData. There's a lot of room for other typed data across chains and such, so we should try to be specific if possible.

File: background/services/keyring/index.ts (hunk 1 above)
Comment (wojciech-turek) [reply background/services/keyring/index.ts:340]: Thanks, I instinctively write arrow functions, changed it to be consistent with other function declarations as suggested.

**Review by Shadowfiend — APPROVED** [review]
Looks good, let's roll :guitar:

**Pull request merged**